cli/lib/pm2-manager.ts

```diff
@@ -0,0 +1,381 @@
+			name: options.name,
+			script: options.script,
+			args: options.args || [],
+			cwd: options.cwd,
+			env: options.env,
+			instances: options.instances || 1,
+			exec_mode: options.exec_mode || 'fork',
+			autorestart: options.autorestart !== false,
+			max_memory_restart: options.max_memory_restart,
+		};
+
+		pm2.start( processConfig, ( error, apps ) => {
+			disconnect();
+			if ( error ) {
+				reject( error );
+				return;
+			}
+
+			if ( ! apps || apps.length === 0 ) {
+				reject( new Error( 'Failed to start process' ) );
+				return;
+			}
+
+			resolve( apps[ 0 ] as ProcessDescription );
+		} );
+	} );
+}
+
+export async function stopProcess( name: string ): Promise< void > {
+	await ensureDaemonRunning();
+
+	return new Promise( ( resolve, reject ) => {
+		pm2.stop( name, ( error ) => {
+			disconnect();
+			if ( error ) {
+				reject( error );
+				return;
+			}
+			resolve();
+		} );
+	} );
+}
+
+export async function deleteProcess( name: string ): Promise< void > {
+	await ensureDaemonRunning();
+
+	return new Promise( ( resolve, reject ) => {
+		pm2.delete( name, ( error ) => {
+			disconnect();
+			if ( error ) {
+				reject( error );
+				return;
+			}
+			resolve();
+		} );
+	} );
+}
+
+export async function restartProcess( name: string ): Promise< void > {
+	await ensureDaemonRunning();
+
+	return new Promise( ( resolve, reject ) => {
+		pm2.restart( name, ( error ) => {
+			disconnect();
+			if ( error ) {
+				reject( error );
+				return;
+			}
+			resolve();
+		} );
+	} );
+}
+
+export async function listProcesses( autoStart = true ): Promise< ProcessDescription[] > {
+	if ( autoStart ) {
+		await ensureDaemonRunning();
+	} else if ( ! isDaemonRunning() ) {
+		return [];
+	} else {
+		await connect();
+	}
+
+	return new Promise( ( resolve, reject ) => {
+		pm2.list( ( error, processes ) => {
+			disconnect();
+			if ( error ) {
+				reject( error );
+				return;
+			}
+			resolve( ( processes || [] ) as ProcessDescription[] );
+		} );
+	} );
+}
+
+export async function describeProcess( name: string ): Promise< ProcessDescription | null > {
+	const processes = await listProcesses();
+	return processes.find( ( p ) => p.name === name ) || null;
+}
+
+export function cleanup(): void {
+	disconnect();
+}
+
+process.on( 'exit', cleanup );
+process.on( 'SIGINT', cleanup );
+process.on( 'SIGTERM', cleanup );
+
+/**
+ * Proxy Server Management Functions
+ *
+ * The proxy runs as a PM2-managed CLI process. When `studio proxy start` is called,
+ * PM2 starts the CLI with the proxy command and keeps it running persistently.
+ */
+
+const PROXY_PROCESS_NAME = 'studio-proxy';
+
+/**
+ * Start the proxy server via PM2
+ * This launches the CLI with `proxy start --managed` which runs the proxy servers
+ */
+export async function startProxyProcess( cliPath: string ): Promise< ProcessDescription > {
+	await ensureDaemonRunning();
+
+	return new Promise( ( resolve, reject ) => {
+		const processConfig: pm2.StartOptions = {
```

Comment:
```suggestion
		const processConfig: Pm2StartOptions = {
```
Doing this and adding an import statement like so at the top seems to work. 

```ts
import { StartOptions as Pm2StartOptions } from 'pm2';
```